### ISSUE TITLE
JavaScript implementation

### ISSUE THREAD

**Issue by evilebottnawi** [jsoref/spelling#3]
What do you think about this?

**Comment by jsoref** [comment]
I had one a decade ago written in xpcshell (Mozilla's SpiderMonkey engine with XPCOM support which allowed for things including file I/o).

I don't know if I still have access to the computers where I ran it.

Practically, there's nothing that requires any particular language. JavaScript's regular expression and string handling is certainly capable of running the same code concepts.

I wouldn't mind hosting a JavaScript implementation, but I probably won't port this anytime soon.

In short, porting `w` and `f` to JavaScript line/expression at a time shouldn't be hard.

**Comment by jsoref** [comment]
@evilebottnawi: out of curiosity, how did you intend to use this?

https://github.com/marketplace/actions/check-spelling is built around this code (using the `composite` host) and can be self-hosted using [nektos/act](https://github.com/nektos/act).
